megamek/src/megamek/server/totalwarfare/TWGameManager.java

```diff
@@ -15512,6 +15523,44 @@ else if ((entity.heat >= 14) && !entity.isShutDown()) {
                             // okay, now mark shut down
                             entity.setShutDown(true);
                         }
+
+                        if (diceRoll.getIntValue() == 2 && riscKit) {
+                            r = new Report(5545);
+                            r.subject(entity.getId());
+                            addReport(r);
+
+                            int hits = 0;
+                            Roll diceRoll2 = Compute.rollD6(2);
+                            r = new Report(6310);
+                            r.subject = entity.getId();
+                            r.add(diceRoll2);
+                            r.newlines = 0;
+                            addReport(r);
+
+                            if ((diceRoll2.getIntValue() == 8) || (diceRoll2.getIntValue() == 9)) {
+                                hits = 1;
+                            } else if ((diceRoll2.getIntValue() == 10) || (diceRoll2.getIntValue() == 11)) {
+                                hits = 2;
+                            } else if (diceRoll2.getIntValue() == 12) {
+                                hits = 3;
+                            }
+
+                            r = new Report(6328);
+                            r.subject = entity.getId();
+                            r.add("%d+1=%d".formatted(hits, hits+1));
+                            addReport(r);
+
+                            hits++;
+
+                            for (int j = 0; (j < 12) && (hits > 0); j++) {
+                                var crit = mek.getCritical(Mek.LOC_CT, j);
+                                if ((crit.getType() == CriticalSlot.TYPE_SYSTEM)
```

Comment:
suggest adding a (crit != null) && ... guard as empty slots will return null. It won't usually make a difference but experience says guard it all the same.

Reply:
I think you can get down to 2 engine slots with a superheavy mech and a compact engine, so it's a realistic concern.

Done.